File: src/jvmMain/java/org/hisp/dhis/lib/expression/spi/VariableValue.java
package org.hisp.dhis.lib.expression.spi;

import java.util.List;

/**
 * A variable value as used int he rule-engine.
 */
public interface VariableValue {

    /**
     * @return variable value, maybe null
     */
    String value();

    Object valueOrDefault();

    /**
     * @return list of candidates, never null
     */
    List<String> candidates();

    /**
     * @return associated event date, maybe null
     */
    String eventDate();

    ValueType valueType();
}

File: src/jvmMain/java/org/hisp/dhis/lib/expression/spi/VariableValue.kt
package org.hisp.dhis.lib.expression.spi

/**
 * A variable value as used int he rule-engine.
 */
interface VariableValue {
    /**
     * @return variable value, maybe null
     */
    fun value(): String?
    fun valueOrDefault(): Any?

    /**
     * @return list of candidates, never null
     */
    fun candidates(): List<String?>?

    /**
     * @return associated event date, maybe null
     */
    fun eventDate(): String?
    fun valueType(): ValueType
}
